pkg/kubectl/cmd/util/factory_object_mapping.go

```diff
@@ -162,14 +158,50 @@ func (f *ring1Factory) Describer(mapping *meta.RESTMapping) (kubectl.Describer,
 			return &kubectl.ClusterDescriber{Interface: fedClientSet}, nil
 		}
 	}
+
+	// make a generic describer if we can, or return the original error
+	makeGenericDescriberOrErr := func(originalErr error) (kubectl.Describer, error) {
+		clientConfig, err := f.clientAccessFactory.ClientConfig()
+		if err != nil {
+			return nil, originalErr
+		}
+		clientConfigCopy := *clientConfig
+		clientConfigCopy.APIPath = dynamic.LegacyAPIPathResolverFunc(mapping.GroupVersionKind)
+		gv := mapping.GroupVersionKind.GroupVersion()
+		clientConfigCopy.GroupVersion = &gv
+
+		// used to look up scope info for the resource
+		discoveryClient, err := f.clientAccessFactory.DiscoveryClient()
+		if err != nil {
+			return nil, originalErr
+		}
+
+		// used to fetch the resource
+		dynamicClient, err := dynamic.NewClient(&clientConfigCopy)
+		if err != nil {
+			return nil, originalErr
+		}
+
+		// used to get events for the resource
+		clientSet, err := f.clientAccessFactory.ClientSet()
+		if err != nil {
+			return nil, originalErr
+		}
+		eventsClient := clientSet.Core()
+
+		return kubectl.GenericDescriberFor(mapping.GroupVersionKind, discoveryClient, dynamicClient, eventsClient), nil
+	}
+
 	clientset, err := f.clientAccessFactory.ClientSetForVersion(&mappingVersion)
 	if err != nil {
-		return nil, err
+		// if this is an unknown group/version, go generic
+		return makeGenericDescriberOrErr(err)
```

Comment:
I see what you are doing here with passing in the error to return, but it wasn't immediately obvious to me.  Is this a pattern we use elsewhere?  Even though there is a bit of code duplication, WDYT of making this look like with how we check for the clientset error?

```go
if err != nil {
  if describer, describerError := makeGenericDescriber(); describeError == nil {
    // If we fail to create a generic describer, return the original error
    return describer, nil
  } else {
    return nil, err
  }
}
```